Sources/MapboxNavigation/RouteLineController.swift

```diff
@@ -103,9 +103,10 @@ extension NavigationMapView {
             if routeLineTracksTraversal {
                 if progress.routeIsComplete && (navigationMapView.routes != nil) {
                     navigationMapView.removeRoutes()
+                } else {
+                    navigationMapView.updateUpcomingRoutePointIndex(routeProgress: progress)
+                    navigationMapView.travelAlongRouteLine(to: location.coordinate)
```

Comment:
I see this code snippet repeats multiple times here. Maybe we could refactor it into a method?

Reply:
Thanks for the suggestion, I'll simplify this part.